File: queue.py
class Queue(object):

    def __init__(self):
        self.queue = []
        self.queue_len = 0

    def push(self, n):
            self.queue_len += 1
            self.queue.insert(0, n)

    def pop(self):
        if not self.is_empty():
            self.queue_len -= 1
            self.queue.pop(self.queue_len)

    def is_empty(self):
        if self.queue_len == 0:
            return True
        return False

queue = Queue()
queue.push(2)   # [2]
queue.push(54)  # [54, 2]
queue.push(1)   # [1, 54, 2]
queue.pop()     # [1, 54]
queue.pop()     # [1]

print queue.queue

File: queue.rb
class Queue
	def initialize
		@queue = []
		@queue_count = 0
	end

	def push(n)
		@queue.push(n)
		@queue_count += 1
	end
	def pop
		return 'Queue Empty' if is_empty
		@queue.shift
		@queue_count -= 1
	end
	def is_empty
		return true if @queue_count == 0
		return false 
	end
	def queue
		return @queue
	end
end

queue = Queue.new
queue.push 2
queue.push 5
queue.push 10
print queue.queue
queue.pop
queue.pop
print queue.queue